lib/iris/tests/unit/fileformats/pp/test__data_bytes_to_shaped_array.py

```diff
@@ -17,10 +17,52 @@
 
 import numpy as np
 import numpy.ma as ma
+import pytest
 
 import iris.fileformats.pp as pp
 
 
+@pytest.mark.parametrize(
+    "data_shape,expected_shape,data_type",
+    [
+        ((2, 3), (2, 3), np.float32),
+        ((2, 3), (3, 2), np.int32),
+        ((2, 3), (1, 3), np.int16),
+        ((2, 3), (2, 2), np.int8),
+        ((2, 3), (3, 3), np.float32),
+        ((2, 3), (2, 4), np.float32),
+    ],
+)
```

Comment:
I feel like changing more than one thing at once defeats the benefit of parameterisation. E.g. how would you tell if different behaviour was caused by `float` vs `int`, or a change in expected shape? Were you trying to save system resources by limiting the number of tests?

I would have assumed it would look something like this:

```suggestion
@pytest.mark.parametrize("data_shape", [(2, 3)])
@pytest.mark.parametrize("expected_shape", [(2, 3), (3, 2), (1, 3), (2, 2), (3, 3), (2, 4)])
@pytest.mark.parametrize("data_type", [np.float32, np.int32, np.int16, np.int8])
```